lib/link_checker.rb

```diff
@@ -0,0 +1,86 @@
+class LinkChecker
+  class << self
+    def extract_urls(data)
+      new(data).extract_urls
+    end
+
+    def link_checker_api_bearer_token
+      ENV["LINK_CHECKER_API_BEARER_TOKEN"]
+    end
+
+    def for_files_at(path)
+      path = File.join(path, "**/*.erb") unless path.to_s.include?("*")
+      data = Dir.glob(path).map { |filename| File.read(filename) }
+      new(data)
+    end
+  end
+
+  attr_reader :data
+
+  def initialize(data)
+    @data = data
+  end
+
+  def urls
+    # Gather urls in a Set to allow simple combination and duplicate removal
+    @urls ||= extract_urls.to_set
```

Comment:
Could extract urls just return a set, instead of having a separate method?

Reply:
If you look at the way it was done in [govuk-coronavirus-find-support](https://github.com/alphagov/govuk-coronavirus-find-support/blob/master/lib/link_checker.rb#L29) you can see doing the `.to_set` outside that method simplifies the code.